src/hyperion/device_setup_plans/setup_panda.py

```diff
@@ -0,0 +1,153 @@
+    """
+
+    sample_velocity_mm_per_s = parameters.x_step_size * 1e-3 / time_between_x_steps_ms
+
+    row1 = SeqTableRow(trigger=SeqTrigger.BITA_1, time2=1)
+    row2 = SeqTableRow(
+        trigger=SeqTrigger.POSA_GT,
+        position=int(parameters.x_start * MM_TO_ENCODER_COUNTS),
+        time2=1,
+        outa1=True,
+        outa2=True,
+    )
+    row3 = SeqTableRow(
+        position=int(
+            (parameters.x_start * MM_TO_ENCODER_COUNTS)
+            + (
+                parameters.x_step_size
+                * (
+                    parameters.x_steps - 1
+                )  # x_start is the first trigger point, so we need to travel to x_steps-1 for the final triger point
+                * MM_TO_ENCODER_COUNTS
+                + (MM_TO_ENCODER_COUNTS * (parameters.x_step_size / 2))
+            )
+        ),
+        trigger=SeqTrigger.POSA_GT,
+        time2=1,
+    )
+
+    row4 = SeqTableRow(trigger=SeqTrigger.BITA_1, time2=1)
+
+    row5 = SeqTableRow(
+        trigger=SeqTrigger.POSA_LT,
+        position=(parameters.x_start * MM_TO_ENCODER_COUNTS)
+        + (
+            parameters.x_step_size * (parameters.x_steps - 1) * MM_TO_ENCODER_COUNTS
+            + (sample_velocity_mm_per_s * exposure_time_s * MM_TO_ENCODER_COUNTS)
+        ),
+        time2=1,
+        outa1=True,
+        outa2=True,
+    )
+
+    row6 = SeqTableRow(
+        trigger=SeqTrigger.POSA_LT,
+        position=parameters.x_start * MM_TO_ENCODER_COUNTS
+        - (MM_TO_ENCODER_COUNTS * (parameters.x_step_size / 2))
+        + (sample_velocity_mm_per_s * exposure_time_s * MM_TO_ENCODER_COUNTS),
+        time2=1,
+    )
+
+    rows = [row1, row2, row3, row4, row5, row6]
+
+    table = seq_table_from_rows(*rows)
+
+    return table
+
+
+def setup_panda_for_flyscan(
+    panda: PandA,
+    config_yaml_path: str,
+    parameters: PandAGridScanParams,
+    initial_x: float,
+    exposure_time_s: float,
+    time_between_x_steps_ms: float,
+) -> MsgGenerator:
+    """This should load a 'base' panda-flyscan yaml file, then set the
+    specific grid parameters, then adjust the PandAsequencer table to match this new grid
+    """
+    yield from load_device(panda, config_yaml_path)
+
+    yield from bps.abs_set(
+        panda.inenc[1].setp, initial_x * MM_TO_ENCODER_COUNTS, wait=True  # type: ignore
+    )
+    LOGGER.info(
+        f"Initialising panda to {initial_x} mm, {initial_x * MM_TO_ENCODER_COUNTS} counts"
+    )
+
+    yield from bps.abs_set(panda.clock[1].period, time_between_x_steps_ms)  # type: ignore
+
+    # The trigger width should last the same length as the exposure time
+    yield from bps.abs_set(panda.pulse[1].width, DETECTOR_TRIGGER_WIDTH)
+
+    table = get_seq_table(parameters, time_between_x_steps_ms, exposure_time_s)
+
+    LOGGER.info(f"Setting Panda sequencer values: {str(table)}")
+
+    yield from bps.abs_set(panda.seq[1].table, table, group="panda-config")
+
+    yield from arm_panda_for_gridscan(
+        panda,
+    )
```

Comment:
Must: If we're able to do this at the same time as the config we should pass that group in here so that we wait on it properly arming in the next line

Reply:
Actually I think we should wait for the table to be set before arming the PandA - if we arm before the table is properly set, there could be situations where we randomly send trigger signals